### PR TITLE
feat(connector-alipay-native): add Alipay Native connector

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add Alipay Native connector.

## Linear Issue Reference

LOG-2454

## Testing

UTs.

### CODE REVIEW

File: packages/connector-alipay-native/src/constant.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { getFileContents } from '@logto/shared';
+
+export const alipayEndpoint = 'https://openapi.alipay.com/gateway.do';
+export const methodForAccessToken = 'alipay.system.oauth.token';
+export const methodForUserInfo = 'alipay.user.info.share';
+
+// Constants in this segment are for SDK call
+// See: https://opendocs.alipay.com/open/218/105327
+export const staticParameters = {
+  apiname: 'com.alipay.account.auth',
```
Comment (demonzoo) [inline packages/connector-alipay-native/src/constant.ts:13]: So alipay has a param called `apiname`?

File: packages/connector-alipay-native/src/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+      .mockResolvedValueOnce(mockedAlipayNativeConfigWithValidPrivateKey);
+    nock(alipayEndpointUrl.origin)
+      .post(alipayEndpointUrl.pathname)
+      .query(true)
+      .reply(200, {
+        alipay_user_info_share_response: {
+          code: '10000',
+          msg: 'Success',
+          user_id: '2088000000000000',
+          nick_name: 'PlayboyEric',
```
Comment (demonzoo) [inline packages/connector-alipay-native/src/index.test.ts:145]: 😂

**Review by demonzoo — COMMENTED** [review]
Looks good to me, but pls wait for a second approver

File: packages/connector-alipay-native/src/constant.ts (hunk 1 above)
Comment (darcyYe) [reply packages/connector-alipay-native/src/constant.ts:13]: Generating URL for native SDK relies on some predefined constants.
You can check [here](https://opendocs.alipay.com/open/218/105327) for more context.

Comment (darcyYe) [reply packages/connector-alipay-native/src/constant.ts:13]: The answer is 'YES'.

Comment (demonzoo) [reply packages/connector-alipay-native/src/constant.ts:13]: SMH

File: packages/connector-alipay-native/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      throw new ConnectorError(ConnectorErrorCodes.General);
+    }
+    // TODO: elaborate on the error messages for all social connectors (see LOG-2157)
+
+    assert(id, new ConnectorError(ConnectorErrorCodes.InvalidResponse));
+
+    return { id, avatar, name };
+  };
+
+  private readonly getTimestamp = (): string => dayjs().format('YYYY-MM-DD HH:mm:ss');
```
Comment (wangsijie) [inline packages/connector-alipay-native/src/index.ts:165]: don't need to be a property? just a util function, can be moved out of class.

**Comment by wangsijie** [comment]
Social Icon 24x24

File: packages/connector-alipay-native/src/constant.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const pathToConfigTemplate = path.join(currentPath, '..', 'docs', 'config-template.md');
+const readmeContentFallback = 'Please check README.md file directory.';
+const configTemplateFallback = 'Please check config-template.md file directory.';
+
+export const defaultMetadata: ConnectorMetadata = {
+  target: 'alipay',
+  type: ConnectorType.Social,
+  platform: ConnectorPlatform.Native,
+  name: {
+    en: 'Sign In with Alipay',
```
Comment (wangsijie) [inline packages/connector-alipay-native/src/constant.ts:29]: Please change this to `Alipay` to align with master.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**